src/chromo/models/qgsjet.py

```diff
@@ -139,7 +139,7 @@ def _cross_section(self, kin=None, full_AA=False):
         """
         kin = self.kinematics if kin is None else kin
 
-        if full_AA and (kin.p1.is_nucleus and kin.p1.A > 1):
+        if max_info and (kin.p1.is_nucleus and kin.p1.A > 1):
```

Comment:
Could target be anything (kin.p2)?

Reply:
Usually the possible targets are restricted by whatever is listed in `_targets`. So there is no explicit need to check this. This function is the correct one to call if the target is heavier than a nucleon or if the projectile is a nucleus. Save for the case below and for dpmjet above.

Reply:
But yeah, actually, there is no check if an externally constructed kinematics object is (without calling `model.kinematics = kin`) is passed